ext/openssl/openssl.c

```diff
@@ -3524,8 +3524,11 @@ PHP_FUNCTION(openssl_csr_sign)
 		goto cleanup;
 	}
 
-
-	ASN1_INTEGER_set(X509_get_serialNumber(new_cert), (long)serial);
+#if defined(PHP_WIN32) && defined(ZEND_ENABLE_ZVAL_LONG64)
+	ASN1_INTEGER_set_int64(X509_get_serialNumber(new_cert), serial);
```

Comment:
Might make sense to simply always call this method? It's unnecessary on 32-bit systems, but I think that's better than doing these platform-specific checks. Or at least I'd make the condition something like `#if SIZEOF_ZEND_LONG == 8`.

Reply:
This function is only [available as of OpenSSL 1.1.0](https://www.openssl.org/docs/man1.1.0/man3/ASN1_INTEGER_set_int64.html#HISTORY), and even master still supports OpenSSL >= 1.0.2 (something we should probably change; [OpenSSL 3 is coming](https://www.openssl.org/blog/blog/2021/06/17/OpenSSL3.0ReleaseCandidate/). We're shipping OpenSSL 1.1 for years on Windows, though.

Reply:
Okay, could we make this OpenSSL version dependent then? Use set_int64 on 1.1 and set below?

Reply:
Right, that makes sense. PR updated.